libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/distanceinfo/PolygonDistanceInfo.kt

```diff
@@ -6,22 +6,24 @@ import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectType
  * PolygonDistanceInfo contains information about distance to the road object represented as
  * polygon.
  *
- * @param distanceToNearestEntry distance to the nearest entry
- * @param distanceToNearestExit distance to the nearest exit
+ * @param entrances distances to particular entrances, sorted by probability.
+ * Might be empty in the case when the polygon is very large and no entry could be detected using most likely path.
+ * @param exits distances to particular exits, sorted by probability.
+ * Might be empty in the case when the polygon is very large and no exit could be detected using most probable path.
  * @param inside true if inside the object
  */
 class PolygonDistanceInfo internal constructor(
     roadObjectId: String,
     @RoadObjectType.Type roadObjectType: Int,
-    val distanceToNearestEntry: Double,
-    val distanceToNearestExit: Double,
+    val entrances: List<Gate>,
+    val exits: List<Gate>,
     val inside: Boolean,
 ) : RoadObjectDistanceInfo(roadObjectId, roadObjectType, RoadObjectDistanceInfoType.POLYGON) {
 
     /**
      * distance to start of the object
      */
-    override val distanceToStart: Double = distanceToNearestEntry
+    override val distanceToStart: Double = entrances.firstOrNull()?.distance ?: 0.0
```

Comment:
For now there are some cases when entrances and exits could be empty. If we enter to a middle of large object, entrances and exits are too far and they couldn't be determined.

I think it's not correct to return zero in such cases, maybe it worths be documented there as well? Another idea is to change `Double` to `Double?` in such cases

Reply:
I'm fine changing it to nullable if you think it's a better approach 👍

Reply:
@0v3rt1r3d BTW this is coming from an `abstract class` https://github.com/mapbox/mapbox-navigation-android/blob/ae3181640aa124a4bc06978d025e8bf28756585c/libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/distanceinfo/RoadObjectDistanceInfo.kt#L29 Is it safe to say that there are also other cases that may be `null`? https://github.com/mapbox/mapbox-navigation-android/tree/main/libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/distanceinfo

Reply:
This is also used in https://github.com/mapbox/mapbox-navigation-android/blob/ae3181640aa124a4bc06978d025e8bf28756585c/libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/UpcomingRoadObject.kt#L19-L22

Reply:
> Is it safe to say that there are also other cases that may be null?

I think no, but maybe it worths to make base class keeping nullable value. Points and gantries are too small and could be always tracked, for polylines we keep distance-tables to avoid such problems

Reply:
Interestingly, we can define the abstract property as nullable and still keep some child classes as non-nullable 🚀 

For example https://github.com/mapbox/mapbox-navigation-android/blob/3a6a23516d9b7ba4c5349201b97d0199fbd999aa/libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/distanceinfo/GantryDistanceInfo.kt#L18

cc @korshaknn